src/RawPOSIX/src/interface/mem.rs

```diff
@@ -226,3 +230,21 @@ pub fn sbrk_handler(cageid: u64, brk: u32) -> i32 {
 
     (PAGESIZE * heap.npages) as i32
 }
+
+pub fn check_and_convert_addr(cageid: u64, user_addr: u64) -> Result<u64, Errno> {
+    let cage = cagetable_getref(cageid);
```

Comment:
@ChinmayShringi all these updates look great, one thing that would be better is moving cagetable_getref outside of this function and just passing the cage struct in. We have to get it again before calling the syscall so itll be better if we just do it once.